sw/device/lib/base/README.md

```diff
@@ -0,0 +1,16 @@
+# `libbase`, the OpenTitan Standard Library
+
+This subtree provides headers and libraries known collectively as `libbase`,
+which serve as OpenTitan's ersatz `libc`.
+
+## Differences from a `libc`
+
+`libbase` is not a `libc`, even though it implements a number of `libc` symbols,
+and should not be used as such.
+`libbase` is, rather, a place for base libraries which may need to make use of
+more compiler/platform intrinsics than average.
+
+In general, a library exposing general utilities for working close to the
+hardware should live in this subtree: for example, a library providing `memcpy`
+and related symbols. A library for talking to a particular peripheral, like a
+UART port, is a non-example.
```

Comment:
@mcy , as per #1187 @gkelly and @lenary  comments, could you please add some more information about the split of libbase. It should be possible to link DIFs against parts of it as opposed to always linking against the whole library. DIFs have some restrictions outlined in the RFC spec, I guess it's worth mentioning that parts of library linked against DIFs must satisfy those requirements.

Reply:
Contrary to the naming, `libbase` is not really a single library. I expect every header of `libbase` to be compiled as a single library, exactly for the reasons you describe. This is consistent with how all other libraries are built under Meson (and most modern build systems, really), so I don't think there's any point to stating it explicitly.